### PR TITLE
Add tablet_max_pending_versions config; improve and bugfix some logs

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [x] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Problem Summary(Required) ：

Make max pending versions of primary key tablet configurable, and increase it to 1000, also refactor and fix some logs

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|2|2|100.00%|

File: be/src/common/config.h [hunk 1]
```diff
@@ -631,6 +631,9 @@ CONF_mInt32(sys_minidump_interval, "600");
 // requests will fail.
 CONF_Int16(tablet_max_versions, "1000");
 
+// The maximum number of pending versions allowed for a primary key tablet
+CONF_mInt32(tablet_max_pending_versions, "10000");
+
```
Comment (Astralidea) [inline be/src/common/config.h:636]: in Problem Summary this value is 1000.

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|2|2|100.00%|

File: be/src/common/config.h [hunk 2]
```diff
@@ -631,6 +631,9 @@ CONF_mInt32(sys_minidump_interval, "600");
 // requests will fail.
 CONF_Int16(tablet_max_versions, "1000");
 
+// The maximum number of pending versions allowed for a primary key tablet
+CONF_mInt32(tablet_max_pending_versions, "1000");
```
Comment (chaoyli) [inline be/src/common/config.h:635]: You can use `tablet_max_versions`, this is the same meaning.

**Review by chaoyli — APPROVED** [review]

**Pull request merged**